src/plugins/intel_gpu/src/graph/include/compilation_context.hpp

```diff
@@ -7,14 +7,15 @@
 #include <threading/ie_cpu_streams_executor.hpp>
 #include <functional>
 #include <memory>
+#include "intel_gpu/graph/kernel_impl_params.hpp"
 
 namespace cldnn {
 
 class ICompilationContext {
 public:
     using Task = std::function<void()>;
-    virtual void push_task(size_t key, Task&& task) = 0;
-    virtual void remove_keys(std::vector<size_t>&& keys) = 0;
+    virtual void push_task(kernel_impl_params key, Task&& task) = 0;
```

Comment:
Can't we make this header to use template and do actual specialization in the cpp code for more general impl of the abstract class?

Reply:
As discussed offline, I'm not sure about how to apply template to the static creation method.